lib/std/setutils.nim

```diff
@@ -0,0 +1,45 @@
+func symmetricDifference*[T](x, y: set[T]): set[T] {.magic: "XorSet".}
+
+func `-+-`*[T](x, y: set[T]): set[T] {.magic: "XorSet".}
+ 
+template toggle*[T: Ordinal](x: var set[T], y: T) =
+  ## Toggles the element `y` in the set `x`.
+  x = x.symmetricDifference({y})
+
+template toggle*[T: Ordinal](x: var set[T], y: set[T]) =
+  ## Toggles the set `y` in the set `x`.
+  x = x.symmetricDifference(y)
+
+# --- set complement ---
+
+func emptySet*[T: Ordinal](x: typedesc[T]): set[T] {.inline.} =
+  ## Returns an empty `set[T]`, same as `var x: set[T] = {}`
+  result = {}
+
+func fullSet*[T: Ordinal](x: typedesc[T]): set[T] {.inline.} =
+  ## Returns a `set[T]` with all elements
+  {T.low .. T.high}
+
+func complement*[T: Ordinal](s: set[T]): set[T] =
+  ## Returns a `set[T]` with the elements that are not inside
+  fullSet(T) - s
+
+func isEmpty*[T: Ordinal](s: set[T]): bool {.inline.} =
+  ## Check if the `set[T]` is empty, same as `s == {}`
+  s == {}
+
+func isFull*[T: Ordinal](s: set[T]): bool {.inline.} =
```

Comment:
are these all in Nim 2 too?

Reply:
those are new ones but it could be backported to Nim 2, the new ones are `emptySet`, `isEmpty`, `isFull`, `[]` and the `toggle` with the T type. some looks pointless but it's better make it quite symmetric (full implies the existence of empty)

Reply:
original nim 2.x `setutils` only has `symmetricDifference`, `-+-`, `toggle`, `fullSet`, `complement` and `[]`

Reply:
Please only add what Nim 2 offers.

Reply:
what if i backport first the new ones to nim 2?

Reply:
That's not how it works. Most API additions require some arguing first or even an RFC.

Reply:
ok so I will remove the new ones to match the setutils of nim 2.x